src/read_csv.py

```diff
@@ -0,0 +1,20 @@
+import csv
+
+
+def read_swaps_from_csv():
+    with open('data/swaps_data_from_router.csv', newline='') as f:
+        reader = csv.reader(f)
+        data = list(reader)
+        orders = dict()
+        for order in data:
+            path_data = order[5].split("'")
+            path_data = path_data[:-1]
+            if(len(path_data) > 1):
+                block_number = int(order[0])
+                list_start = orders[block_number] if block_number in \
+                    orders else list()
+                list_start.append(
+                    {"sellToken": path_data[1],
+                     'buyToken': path_data[-1]})
```

Comment:
I'm not sure about this one - haven't you removed the last element of the path in line 11
```            path_data = path_data[:-1]```
so the element that you're accessing here is just an intermediate token?

Reply:
Yeah, I removed the last element, which was just an "]", due to my unclean parsing. Sorry for the ugly code. I was definitively in a rush yesterday. Marcos code is much cleaner, now. But likely, I will need to do more cleanups.